From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Uti Edun <uti.edun@chainguard.dev>
Date: Sat, 4 Oct 2025 19:30:00 +0000
Subject: [PATCH] Consolidated fixes for Ceph 18.2.7

This patch includes:
1. pybind/rbd: use libcpp.bool for librbd boolean arguments
2. pybind/rgw: use libcpp.bool for rgw boolean arguments
3. rgw: add 4th argument to set_parquet_object call

---
 src/pybind/rbd/c_rbd.pxd    | 6 +++---
 src/pybind/rbd/mock_rbd.pxi | 6 +++---
 src/pybind/rbd/rbd.pyx      | 6 ++++--
 src/pybind/rgw/rgw.pyx      | 3 ++-
 src/rgw/rgw_s3select.cc     | 3 ++-
 5 files changed, 14 insertions(+), 10 deletions(-)

diff --git a/src/pybind/rbd/c_rbd.pxd b/src/pybind/rbd/c_rbd.pxd
index 9624cb4ce05..4bce3846eea 100644
--- a/src/pybind/rbd/c_rbd.pxd
+++ b/src/pybind/rbd/c_rbd.pxd
@@ -2,6 +2,7 @@

 from libc.stdint cimport *
 from ctime cimport time_t, timespec
+cimport libcpp

 cdef extern from "rados/librados.h":
     enum:
@@ -530,7 +531,7 @@ cdef extern from "rbd/librbd.h" nogil:
     int rbd_snap_unprotect(rbd_image_t image, const char *snap_name)
     int rbd_snap_is_protected(rbd_image_t image, const char *snap_name,
                               int *is_protected)
-    int rbd_snap_exists(rbd_image_t image, const char *snapname, bint *exists)
+    int rbd_snap_exists(rbd_image_t image, const char *snapname, libcpp.bool *exists)
     int rbd_snap_get_limit(rbd_image_t image, uint64_t *limit)
     int rbd_snap_set_limit(rbd_image_t image, uint64_t limit)
     int rbd_snap_get_timestamp(rbd_image_t image, uint64_t snap_id, timespec *timestamp)
@@ -721,9 +722,9 @@ cdef extern from "rbd/librbd.h" nogil:
     int rbd_namespace_list(rados_ioctx_t io, char *namespace_names,
                            size_t *size)
     int rbd_namespace_exists(rados_ioctx_t io, const char *namespace_name,
-                             bint *exists)
+                             libcpp.bool *exists)

-    int rbd_pool_init(rados_ioctx_t, bint force)
+    int rbd_pool_init(rados_ioctx_t, libcpp.bool force)

     void rbd_pool_stats_create(rbd_pool_stats_t *stats)
     void rbd_pool_stats_destroy(rbd_pool_stats_t stats)
diff --git a/src/pybind/rbd/mock_rbd.pxi b/src/pybind/rbd/mock_rbd.pxi
index 04a056fcc64..dd9596bd310 100644
--- a/src/pybind/rbd/mock_rbd.pxi
+++ b/src/pybind/rbd/mock_rbd.pxi
@@ -2,6 +2,7 @@

 from libc.stdint cimport *
 from ctime cimport time_t, timespec
+cimport libcpp

 cdef nogil:
     enum:
@@ -642,7 +643,7 @@ cdef nogil:
     int rbd_snap_is_protected(rbd_image_t image, const char *snap_name,
                               int *is_protected):
         pass
-    int rbd_snap_exists(rbd_image_t image, const char *snapname, bint *exists):
+    int rbd_snap_exists(rbd_image_t image, const char *snapname, libcpp.bool *exists):
         pass
     int rbd_snap_get_limit(rbd_image_t image, uint64_t *limit):
         pass
@@ -907,9 +908,9 @@ cdef nogil:
                            size_t *size):
         pass
     int rbd_namespace_exists(rados_ioctx_t io, const char *namespace_name,
-                             bint *exists):
+                             libcpp.bool *exists):
         pass
-    int rbd_pool_init(rados_ioctx_t io, bint force):
+    int rbd_pool_init(rados_ioctx_t io, libcpp.bool force):
         pass
     void rbd_pool_stats_create(rbd_pool_stats_t *stats):
         pass
diff --git a/src/pybind/rbd/rbd.pyx b/src/pybind/rbd/rbd.pyx
index 10464821b63..e0378b03674 100644
--- a/src/pybind/rbd/rbd.pyx
+++ b/src/pybind/rbd/rbd.pyx
@@ -23,6 +23,7 @@ from libc cimport errno
 from libc.stdint cimport *
 from libc.stdlib cimport malloc, realloc, free
 from libc.string cimport strdup, memset
+cimport libcpp

 try:
     from collections.abc import Iterable
@@ -1952,7 +1953,7 @@ class RBD(object):
         cdef:
             rados_ioctx_t _ioctx = convert_ioctx(ioctx)
             const char *_name = name
-            bint _exists = False
+            libcpp.bool _exists = False
         with nogil:
             ret = rbd_namespace_exists(_ioctx, _name, &_exists)
         if ret != 0:
@@ -1995,7 +1996,7 @@ class RBD(object):
         """
         cdef:
             rados_ioctx_t _ioctx = convert_ioctx(ioctx)
-            bint _force = force
+            libcpp.bool _force = force
         with nogil:
             ret = rbd_pool_init(_ioctx, _force)
         if ret != 0:
@@ -3707,7 +3708,7 @@ cdef class Image(object):
         name = cstr(name, 'name')
         cdef:
             char *_name = name
-            bint _exists = False
+            libcpp.bool _exists = False
         with nogil:
             ret = rbd_snap_exists(self.image, _name, &_exists)
         if ret != 0:
diff --git a/src/pybind/rgw/rgw.pyx b/src/pybind/rgw/rgw.pyx
index 9bbcdfff586..d210a70bbb8 100644
--- a/src/pybind/rgw/rgw.pyx
+++ b/src/pybind/rgw/rgw.pyx
@@ -7,6 +7,7 @@ from cpython cimport PyObject, ref, exc, array
 from libc.stdint cimport *
 from libc.stdlib cimport malloc, realloc, free
 from cstat cimport stat
+cimport libcpp

 IF BUILD_DOC:
     include "mock_rgw.pxi"
@@ -373,7 +374,7 @@ cdef class LibRGWFS(object):
         cdef:
             rgw_file_handle *_dir_handler = <rgw_file_handle*>dir_handler.handler
             uint64_t _offset = offset
-            bint _eof
+            libcpp.bool _eof
             uint32_t _flags = flags
         with nogil:
             ret = rgw_readdir(self.fs, _dir_handler, &_offset, &readdir_cb,
diff --git a/src/rgw/rgw_s3select.cc b/src/rgw/rgw_s3select.cc
index 1b7dced2782..e3b24f1c62e 100644
--- a/src/rgw/rgw_s3select.cc
+++ b/src/rgw/rgw_s3select.cc
@@ -481,7 +481,8 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_parquet(const char* query)
     try {
       //at this stage the Parquet-processing requires for the meta-data that reside on Parquet object
-      m_s3_parquet_object.set_parquet_object(std::string("s3object"), &s3select_syntax, &m_rgw_api);
+      m_s3_parquet_object.set_parquet_object(std::string("s3object"), &s3select_syntax,
+                                            &m_rgw_api, parquet_object::csv_definitions{});
     } catch(base_s3select_exception& e) {
       ldpp_dout(this, 10) << "S3select: failed upon parquet-reader construction: " << e.what() << dendl;
       fp_result_header_format(m_aws_response_handler.get_sql_result());
--
2.51.0
